tests/cached_itarbundle.rs

```diff
@@ -207,34 +205,54 @@ where
             .join("assets");
         TarIndex::from_dir(root).unwrap()
     })));
+
+    let (url_available_tx, url_available_rx) = std::sync::mpsc::channel();
+    let (server_shutdown_tx, server_shutdown_rx) = futures::channel::oneshot::channel::<()>();
     let tar_service_clone = Arc::clone(&tar_service);
 
-    let server = Server::bind(&addr).serve(move || {
-        let tar_service = Arc::clone(&tar_service_clone);
-        service_fn(move |req| tar_service.response(req))
-    });
+    let server_thread = thread::spawn(move || {
+        let tar_service = tar_service_clone;
 
-    // server is listening now
-    tar_service.set_local_addr(server.local_addr());
-    let url = tar_service.url();
+        let rt = runtime::Builder::new_current_thread()
+            .enable_io()
+            .build()
+            .unwrap();
+
+        let tar_service_clone = Arc::clone(&tar_service);
+        rt.block_on(async move {
+            let server = Server::bind(&addr).serve(make_service_fn(move |_| {
```

Comment:
This got moved into the thread because tectonic uses the reqwest blocking client, which is incompatible with an outer runtime executing at the same time, but also needs a client active at the time of call to `Server::bind`. The fundamental issue is that we're mixing an async library (`hyper`) with a non-async one (`tectonic`) that is using an async one (`reqwest`). A 'real' fix would be to make tectonic `async`, but that seems... difficult, and possibly undesirable.